### PR TITLE
Upgrade agent-test-tools and polish dubbo plugin

### PR DESCRIPTION
**Pull request by nisiyong**
* Upgrade skywalking-agent-test-tools
* Update docs and fix wrong spelling
* Relate https://github.com/apache/skywalking-java/pull/382
* Relate https://github.com/apache/skywalking-agent-test-tool/pull/47

- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
@nisiyong It seems the tool change breaks.

**Comment by nisiyong** [comment]
> @nisiyong It seems the tool change breaks.

I will take a look.

File: test/plugin/agent-test-tools/pom.xml [hunk 1]
```diff
@@ -35,9 +35,9 @@
     <packaging>pom</packaging>
 
     <properties>
-        <agent-test-tools.version>7f20775e0631356c4823d9372b09d653db0e6540</agent-test-tools.version>
+        <agent-test-tools.version>8a09b6f16b0b3fb535ca515437ef107042aaae45</agent-test-tools.version>
         <agent-test-tools.workingDirectory>${project.basedir}/target/agent-test-tools</agent-test-tools.workingDirectory>
-        <agent-test-tools.repos>https://github.com/apache/skywalking-agent-test-tool.git</agent-test-tools.repos>
+        <agent-test-tools.repos>https://github.com/nisiyong/skywalking-agent-test-tool.git</agent-test-tools.repos>
```
Comment (nisiyong) [inline test/plugin/agent-test-tools/pom.xml:40]: @wu-sheng After https://github.com/apache/skywalking-agent-test-tool/pull/49 merge, I will change the repo URL.

**Comment by nisiyong** [comment]
Finally, the agent-test-tools has fixed the prefix bug. And  at the same time, I found  `start with` or `not blank` couldn't use at the MeterItems tag value, because the MeterId will compare with the MeterItems tag value.

[MeterId.java](https://github.com/apache/skywalking-agent-test-tool/blob/master/validator/src/main/java/org/apache/skywalking/plugin/test/agent/tool/validator/entity/MeterId.java)

```java
public class MeterId {
    ...
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        MeterId meterId = (MeterId) o;
        return Objects.equals(name, meterId.name) &&
            Objects.equals(tags, meterId.tags);
    }
}
```

So I am not going to change the dubbo thread pool name, but the agent-test-tools still support the new operators `start with` and `end with` in other cases.

**Comment by wu-sheng** [comment]
I am not very clear why `eq` works, but `start with` not?

**Comment by nisiyong** [comment]
> I am not very clear why `eq` works, but `start with` not?

All the operator could work well at the expected value. But in the meter scenario, the test-tools will find the same meter first, then compare the meter value. The same meters must have the same tag-keys and the same tag-value, that's why `MeterId#equals` compare with tags and `MeterId.Tag#equals` compared with tag-key and tag-value.

**Comment by nisiyong** [comment]
The agent-test -tools repo URL and revision has be updated. All things are done, @wu-sheng please take a look.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**